pkgs/by-name/ed/edk2/package.nix

```diff
@@ -48,29 +46,27 @@ edk2 = stdenv.mkDerivation rec {
     })
   ];
 
-  srcWithVendoring = fetchFromGitHub {
+  src = fetchFromGitHub {
     owner = "tianocore";
     repo = "edk2";
     rev = "edk2-stable${edk2.version}";
     fetchSubmodules = true;
-    hash = "sha256-Nurm6QNKCyV6wvbj0ELdYAL7mbZ0yg/tTwnEJ+N18ng=";
+    hash = "sha256-7vNodHocwqQiO0ZXtqo8lEOFyt8JkFHcAathEhrKWE0=";
+
+    # We don't want EDK2 to keep track of OpenSSL,
+    # they're frankly bad at it.
+    postFetch = ''
+      rm -rf $out/CryptoPkg/Library/OpensslLib/openssl
+      mkdir -p $out/CryptoPkg/Library/OpensslLib/openssl
+      tar --strip-components=1 -xf ${buildPackages.openssl.src} -C $out/CryptoPkg/Library/OpensslLib/openssl
+
+      # Fix missing INT64_MAX include that edk2 explicitly does not provide
+      # via it's own <stdint.h>. Let's pull in openssl's definition instead:
+      sed -i $out/CryptoPkg/Library/OpensslLib/openssl/crypto/property/property_parse.c \
```

Comment:
This maybe look a bit simpler, but don't we have now to re-fetch everytime, we want to patch anything? An alternative would be to move modifications to patchPhase instead.

Reply:
Thoughts on https://github.com/NixOS/nixpkgs/pull/321143/commits/8cd9620f0d7bfc76ed4707978b26af0d6889ce16 ?

Reply:
I think OpenSSL needs to be replaced in src since the src derivation is reused below in edk2.mkDerivation

Reply:
I can do some more testing. My goal was to fix the automatic updating

Reply:
Yeah, `nix-update` doesn't know how to handle the source hash not being in `src`. 

```
nix_update.errors.UpdateError: Could not find a url in the derivations src attribute
```

I think our options are either:

1. Move back to the separate `runCommand` wrapper we had before this PR, and write a custom update script
2. Accept that we'll have to re-fetch the source if we want to modify the content in that previous `runCommand` (e.g. the openssl replacement I put in `postFetch` and you moved to `postPatch`)

Reply:
That doesn't work @Mic92, OVMF will use edk2.src, not the edk2.src post patched.

Reply:
I've removed your commit @Mic92 as it won't work as expected.